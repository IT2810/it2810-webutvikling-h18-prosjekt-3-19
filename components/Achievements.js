import React, { Component } from 'react';
import {
	View,
	Text,
	StyleSheet,
	ScrollView,
	Image,
} from 'react-native';

export default class Achievements extends Component {
	constructor(props){
		super(props);
		this.state = {
			daily: 20,
		};
	}
	render() {
		const five = '../assets/images/achievements/5.png';
		const five_ = '../assets/images/achievements/5_.png';
		
		return (
				<View style={styles.container}>

					<Text style={styles.header}>Daily Completed</Text>
					<Text style={styles.totaltext}>Best: {this.state.daily} completed todos in day</Text>

					<View style={styles.badgeContainer}>
						<Image
							style={styles.achievementBadge}
							source={
								this.state.daily >= 5
				          ? require('../assets/images/achievements/5.png')
				          : require('../assets/images/achievements/5_.png')
							}
						/>
						<Image
							style={styles.achievementBadge}
							source={
								this.state.daily >= 10
				          ? require('../assets/images/achievements/10.png')
				          : require('../assets/images/achievements/10_.png')
							}
						/>
						<Image
							style={styles.achievementBadge}
							source={
								this.state.daily >= 20
				          ? require('../assets/images/achievements/20.png')
				          : require('../assets/images/achievements/20_.png')
							}
						/>
						<Image
							style={styles.achievementBadge}
							source={
								this.state.daily >= 30
				          ? require('../assets/images/achievements/30.png')
				          : require('../assets/images/achievements/30_.png')
							}
						/>
						<Image
							style={styles.achievementBadge}
							source={
								this.state.daily >= 40
				          ? require('../assets/images/achievements/40.png')
				          : require('../assets/images/achievements/40_.png')
							}
						/>
						<Image
							style={styles.achievementBadge}
							source={
								this.state.daily >= 50
				          ? require('../assets/images/achievements/50.png')
				          : require('../assets/images/achievements/50_.png')
							}
						/>
					</View>
				</View>
		);
	}

}
const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingTop: 15,
		paddingLeft: 15,
		paddingRight: 15,
	},
	header: {
		fontSize: 25,
	},
	totaltext: {
		fontSize: 10,
		marginTop:10,
		textAlign: 'right',
	},
	achievementBadge: {
		margin:20,
		width:50,
		height:50,
	},
	badgeContainer: {
		flex: 1,
		flexDirection: 'row',
		flexWrap: 'wrap',
		justifyContent: 'space-between',
	}
});
