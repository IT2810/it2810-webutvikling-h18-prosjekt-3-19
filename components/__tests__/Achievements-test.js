import React from "react";
import Achievements from "../Achievements";
import renderer from "react-test-renderer";

test("Achievements renders correctly", () => {
  const tree = renderer.create(<Achievements />);
  expect(tree.toJSON()).toMatchSnapshot();
});
