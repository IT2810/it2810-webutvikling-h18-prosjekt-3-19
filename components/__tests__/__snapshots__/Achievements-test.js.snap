// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievements renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 25,
      }
    }
  >
    Daily Completed
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 10,
        "marginTop": 10,
        "textAlign": "right",
      }
    }
  >
    Best: 
    20
     completed todos in day
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "margin": 20,
          "width": 50,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "margin": 20,
          "width": 50,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "margin": 20,
          "width": 50,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "margin": 20,
          "width": 50,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "margin": 20,
          "width": 50,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "margin": 20,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;
